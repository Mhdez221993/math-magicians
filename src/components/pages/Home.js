import React from 'react';

const Home = () => (
  <div className="description">
    <h2>Welcome to the Math Magicians App</h2>
    <p>
      Calculator provides simple and advanced mathematical functions in a beautifully designed app.
      Perform basic calculations such as addition, subtraction, multiplication, and division
      Do scientific operations such as trigonometric, logarithmic, and exponential functions
    </p>
  </div>
);
export default Home;
