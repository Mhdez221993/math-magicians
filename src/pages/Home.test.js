import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('test calculator and correct rendering of the Home page', () => {
  it('Home page is rendered correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('Test Home title', () => {
    const { getByText } = render(<Home />);

    expect(getByText('Welcome to the Math Magicians App')).toBeInTheDocument();
  });
});
