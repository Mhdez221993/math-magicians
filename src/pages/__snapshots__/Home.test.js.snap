// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test calculator and correct rendering of the Home page Home page is rendered correctly 1`] = `
<div
  className="description"
>
  <h2>
    Welcome to the Math Magicians App
  </h2>
  <p>
    Calculator provides simple and advanced mathematical functions in a beautifully designed app. Perform basic calculations such as addition, subtraction, multiplication, and division Do scientific operations such as trigonometric, logarithmic, and exponential functions
  </p>
</div>
`;
